kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/EnvelopeEncryptionFilter.java

```diff
@@ -92,7 +92,7 @@ private CompletionStage<ProduceRequestData> maybeEncodeProduce(ProduceRequestDat
                                     ppd.index(),
                                     new EncryptionScheme<>(kekId, EnumSet.of(RecordField.RECORD_VALUE)),
                                     encryptionRequests,
-                                    (kafkaRecord, encryptedValue, headers) -> builder.append(kafkaRecord.timestamp(), kafkaRecord.key(), encryptedValue, headers))
+                                    (kafkaRecord, encryptedValue, headers) -> builder.appendWithOffset(kafkaRecord.offset(), kafkaRecord.timestamp(), kafkaRecord.key(), encryptedValue, headers))
```

Comment:
On the produce path, I'm not sure if using `#append()` would actually cause a defect.  It is the broker that assigns the offset to the record, so I *think* the offset in the record on the produce path will be ignored anyway.  However, I think it will keep the world simpler if we retain the offset from producer -> broker.  This has the advantage of symmetry with the fetch path, and will help keep debugging/tracing easy.